packages/shared/src/components/bookmark/BookmarkFolderContextMenu.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ReactElement, useId } from 'react';
+import { useRouter } from 'next/router';
+import { BookmarkFolder } from '../../graphql/bookmarks';
+import useContextMenu from '../../hooks/useContextMenu';
+import { usePrompt } from '../../hooks/usePrompt';
+import OptionsButton from '../buttons/OptionsButton';
+import { ButtonSize, ButtonVariant } from '../buttons/common';
+import ContextMenu from '../fields/ContextMenu';
+import { EditIcon, TrashIcon } from '../icons';
+import { useLazyModal } from '../../hooks/useLazyModal';
+import { LazyModal } from '../modals/common/types';
+import { useBookmarkFolder } from '../../hooks/bookmark/useBookmarkFolder';
+
+interface BookmarkFolderContextMenuProps {
+  folder: BookmarkFolder;
+}
+
+export const BookmarkFolderContextMenu = ({
+  folder,
+}: BookmarkFolderContextMenuProps): ReactElement => {
+  const contextMenuId = useId();
+  const router = useRouter();
+  const { openModal, closeModal } = useLazyModal();
+  const { isOpen, onMenuClick } = useContextMenu({ id: contextMenuId });
+  const { showPrompt } = usePrompt();
+  const { update: updateFolder, delete: deleteFolder } = useBookmarkFolder({
+    id: folder?.id,
+  });
+
+  const handleDelete = async () => {
+    const confirm = await showPrompt({
+      title: `Delete ${folder.name}?`,
+      description: 'This will also delete all bookmarks in this folder.',
+      cancelButton: { title: 'Cancel' },
+      okButton: { title: 'Delete folder' },
+    });
+
+    if (confirm) {
+      await router.push('/bookmarks');
+      await deleteFolder.mutate(folder.id);
```

Comment:
Should it probably be the other way around?
```suggestion
      await deleteFolder.mutate(folder.id);
      await router.push('/bookmarks');
```

Reply:
I've tried, but page glitch too much because we have an empty page at start (probably due to optimistic update) then redirect 🤔